tests/testsuite/cargo_command.rs

```diff
@@ -391,10 +391,13 @@ fn cargo_subcommand_env() {
         .canonicalize()
         .unwrap();
     let envtest_bin = envtest_bin.to_str().unwrap();
+    // Previously, `$CARGO` would be left at `envtest_bin`. However, with the
```

Comment:
Seeing this test change and I wonder if using `$CARGO` as a Cargo wrapper is a supported use case, and we don't want to reset it. 

Maybe we can detect iff the current `$CARGO` is `cargo[.exe]` and then we reset?

Reply:
cc @epage as you proposed the current solution

Reply:
I have no problem with implementing this is @epage agrees.

Reply:
`CARGO` was originally not intended for purposes of being a wrapper.  That people could do it was a byproduct of the implementation that was done but it wasn't intentional.  I wouldn't worry too much about it.

Reply:
That is fair. Thanks Ed.